src/wifi/model/dmg-ap-wifi-mac.cc

```diff
@@ -1301,7 +1301,9 @@ DmgApWifiMac::StartDataTransmissionInterval (void)
                 }
             }
           else if ((field.GetAllocationType () == CBAP_ALLOCATION) &&
-                  ((field.GetSourceAid () == AID_BROADCAST) || (field.GetSourceAid () == AID_AP) || (field.GetDestinationAid () == AID_AP)))
+                  (((field.GetSourceAid () == AID_BROADCAST) && AccessAllowedInBroadcastCbap (AID_AP)) ||
```

Comment:
could you explain this?

Reply:
Basically, in this part of `DmgApWifiMac::StartDataTransmissionInterval` (same for `DmgStaWifiMac::StartDataTransmissionInterval`) if the allocation is a CBAP, an STA can access the contention period if it's destination or source of the CBAP or the CBAP is broadcast (i.e. the source AID is the broadcast AID). In the latter case, I added the check on the accessibility of the broadcast CBAP by the STA.